### PR TITLE
Added KeyringFactoryWrapper as a member variable of KeyringContainer

### PR DESCRIPTION
**Pull request by aeharvlee**
## Proposed changes

This PR improves SDK Usability for developers using Caver SDK.

Improving SDK usability means providing users with a similar development experience 
no matter what programming language the SDK is implemented in.

**1. Add `KeyringFactoryWrapper` class and add it as a member variable of `KeyringContainer` class**

KeyringFactoryWrapper is a wrapper class which wraps all of static methods of KeyringFactory as public methods.

By adding this, developers can experience the same usability 
no matter what programming language the SDK is implemented in.
e.g., You can generate SingleKeyring by using `caver.wallet.keyring.generate()` in `caver-js` and `caver-java` both.

I also updated test cases related with this PR to KeyringContainerTest.

**2. Chore**

Apply formatting code and import optimization features of IDE.
Import Optimization remove unused package import statements.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-java/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-java)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core/src/main/java/com/klaytn/caver/Caver.java [hunk 1]
```diff
@@ -32,15 +31,15 @@
     public static String DEFAULT_URL = "http://localhost:8551";
 
     /**
-     * @deprecated Please use <code>caver.rpc.klay</code> instead.
      * @see RPC#klay
+     * @deprecated Please use <code>caver.rpc.klay</code> instead.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/Caver.java:35]: If the changes are not essential, please keep the existing code.

File: core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java [hunk 2]
```diff
@@ -40,11 +41,19 @@
...
+     * which means giving developers similar development experience with caver-js
+     */
+    public KeyringFactoryWrapper keyring;
 
     /**
      * Creates KeyringContainer instance
      */
-    public KeyringContainer() {}
+    public KeyringContainer() {
+        keyring = new KeyringFactoryWrapper();
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java:55]: This should be applied in other constructors as well.

File: core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package com.klaytn.caver.wallet.keyring;
+
+import org.web3j.crypto.CipherException;
+
+import java.io.IOException;
+import java.util.List;
+
+public class KeyringFactoryWrapper {
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java:8]: Please add description for javadoc.

File: core/src/main/java/com/klaytn/caver/Caver.java (hunk 1 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/Caver.java:35]: @jimni1222 

Yeap. This is not essential though.
It looks better not to change it.

I will reflect your feedbacks and revert it.

File: core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java (hunk 2 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java:55]: I missed it 😿 
I'll add initialization to another constructor also

File: core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java (hunk 3 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java:8]: Ok I will 👍 .
Thanks for your feedback.

Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java:27]: @jimni1222 
I added descriptions here +_+
[Add descriptions to KeyringFactoryWrapper](https://github.com/klaytn/caver-java/pull/215/commits/6ec62449d882913846c787e54d89c6e2bdb2ee54)

File: core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java (hunk 2 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/wallet/KeyringContainer.java:55]: I applied your feedback here ^_< [Add missed member variable initialization](https://github.com/klaytn/caver-java/pull/215/commits/75a2b2114b3fafd3951a0ed1ed73732462b3908d)

File: core/src/main/java/com/klaytn/caver/Caver.java (hunk 1 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/Caver.java]: I revert it with [Revert code formatting and import optimization for Caver](https://github.com/klaytn/caver-java/pull/215/commits/d45909eb60a5cfa78cd45b9f0489cc00624f6b9b)

File: core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2020 The caver-java Authors
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java:2]:
```suggestion
 * Copyright 2021 The caver-java Authors
```

File: core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+    /**
+     * Creates a KeyringFactoryWrapper instance
+     */
+    public KeyringFactoryWrapper() {
+    }
+
+    /**
+     * Generates a keyring instance.
+     * @return Keyring
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java:37]: I saw that KeyringFactory's return also says Keyring.
Can you change the return value in more detail at the other PR? (`KeyringFactoryWrapper.java` and `KeyringFactory.java`)

File: core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java [hunk 6]
```diff
@@ -108,118 +111,113 @@ public void checkAddress(List<String> expectedAddress, KeyringContainer containe
...
         public void newRoleBasedKeyring() {
-            KeyringContainer container = new KeyringContainer();
-            String expectAddress = PrivateKey.generate().getDerivedAddress();
+            String expectAddress = caver.wallet.keyring.generate().getAddress();
             String[][] expectPrivateKeyArr = {
                     {
-                        PrivateKey.generate().getPrivateKey(),
-                        PrivateKey.generate().getPrivateKey(),
-                        PrivateKey.generate().getPrivateKey(),
+                            caver.wallet.keyring.generate().getKey().getPrivateKey(),
```
Comment (jimni1222) [inline core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java:163]: Can you check indentation here?

File: core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java [hunk 7]
```diff
@@ -228,14 +226,16 @@ public void updateToCoupledKeyring() {
         //CA-KEYRINGCONTAINER-008
         @Test
         public void updateToDecoupledKeyring() {
-            KeyringContainer container = new KeyringContainer();
-            SingleKeyring coupled = KeyringFactory.generate();
-            SingleKeyring deCoupled = KeyringFactory.createWithSingleKey(coupled.getAddress(), PrivateKey.generate().getPrivateKey());
+            SingleKeyring coupled = caver.wallet.keyring.generate();
+            SingleKeyring deCoupled = caver.wallet.keyring.createWithSingleKey(
+                    coupled.getAddress(),
+                    caver.wallet.keyring.generate().getKey().getPrivateKey()
```
Comment (jimni1222) [inline core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java:232]: I think you can use `caver.wallet.keyring.generateSingleKey(1)` here :)

File: core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java (hunk 6 above)
Comment (sirano11) [inline core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java:163]: Why you use a caver.wallet.keyring.generate()?
caver.wallet.keyring.generate() makes a SingleKeyrinng instance.

Comment (aeharvlee) [reply core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java:163]: I think more proper way to handle this is to use `caver.wallet.keyring.generateSingleKey()`.

In [caver-js/test/packages/caver.wallet#L184](https://github.com/klaytn/caver-js/blob/ae1b51295614131c364b1e3e16a82a98bb629e2e/test/packages/caver.wallet.js#L184), it creates a private key string by calling `caver.wallet.keyring.generateSingleKey()` which is also implemented in caver-java.

@sirano11 
If you are agree with this, I'll change `caver.wallet.keyring.generate()` to `caver.wallet.keyring.generateSingleKey()`.

Thanks for your feedback.

File: core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java (hunk 7 above)
Comment (aeharvlee) [reply core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java:232]: That would be more great :)

I checked [caver-js/test/packages/caver.wallet#L184](https://github.com/klaytn/caver-js/blob/ae1b51295614131c364b1e3e16a82a98bb629e2e/test/packages/caver.wallet.js#L184) is creating a private key string by calling `caver.wallet.keyring.generateSingleKey()` which is also implemented in caver-java.

Thanks for your feedback.

File: core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java (hunk 6 above)
Comment (aeharvlee) [reply core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java]: I checked indentations of `core/src/test/java/com/klaytn/caver/common/KeyringContainerTest.java` 
and had some talks with @sirano11 about what indent will we use.

And we agreed to use indent "4" not "8" (even at Continuation Indent).

So I pushed another [commit](https://github.com/klaytn/caver-java/pull/215/commits/e8f08714b4ff88fb8038a0698a20d16daa924b7d) here to apply indent 4 😃

Thanks for feedback @jimni1222

File: core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java (hunk 5 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/wallet/keyring/KeyringFactoryWrapper.java]: Hi @jimni1222  
@sirano11 and I discussed about this issue and we decide to update return type of comments of `KeyringFactory` and `KeyringFactoryWrapper`.

Here is the related commit ["Update comments of methods of KeyringFactory"](https://github.com/klaytn/caver-java/pull/215/commits/284188bdec3acc84a568216f6d17e616cf24e6a7) I pushed :)

Thanks for your feedback.

**Review by jimni1222 — APPROVED** [review]
LGTM =) Thank you for changing! And please merge this PR after caver-java v1.6.0 is released.

**Review by sirano11 — APPROVED** [review]

**Pull request merged**